### ISSUE TITLE
Algorithmic Regularization in Over-parameterized Matrix Sensing and Neural Networks with Quadratic Activations

### ISSUE THREAD

**Issue by ktrmnm** [ktrmnm/colt18_yomi#1]
- Yuanzhi Li, Tengyu Ma, Hongyang Zhang
- [PMLR](http://proceedings.mlr.press/v75/li18a.html)

## 概要

オーバーパラメトライズされた行列分解モデル，および活性化関数が2次の3層ニューラルネットワークにおいて，gradient descentが暗黙の正則化を行なっていることを明らかにした．

具体的には次のことを示した： $n$ 個のlinear measurementから，ランク $r$ の $d \times d$ p.s.d. 行列 $X^\*$ を復元する問題を考える．このとき，$X^\*$ を $UU^\top$ ($U \in \mathbb{R}^{d \times d}$) とパラメトライズすることによって，$r \ll d$ であっても，$n = \tilde{O}(dr^2)$ 個の観測から復元可能である．また，gradient descentのイテレーション数はおよそ $\tilde{O}(\sqrt{r})$ である．

## 読んだ度 (1: アブスト, 2: 主定理の主張, 3: 証明技術)
2

### 設定

- $X^\*$ は未知のp.s.d行列とする
- $m$ 個のlinear measurement $y\_1, \ldots, y\_m$ が得られるとする．ただし，linear measurementとは，既知の対称行列 $A$ があって $y = \langle A, X^\* \rangle$ (成分ごとの積の和) という観測が得られるということ．
- $X^\*$ の推定量を $UU^\top$ とする．$U$ は $d \times d$ 行列である．
- 経験損失 $f(U) = (2m)^{-1} \sum_{i=1}^m (y_i - \langle A_i, UU^\top \rangle)^2$

パラメータ数は $d^2$ である．$d^2 > m$ であれば潜在的にはオーバーフィットの可能性がある．そこで，gradient descentの局所解はオーバーフィットを避けた好ましい性質を持つということを何らかの意味で言いたい．

### Theorem 1.1

アルゴリズムの仮定
- Gradient descentの反復を $T$ 回で止めるとする
- 初期値は $\alpha B$ とする．ただし $B \in O(d)$ は任意の正規直交行列
- 学習率は $\eta$ とする．

観測の仮定
- $A\_1, \ldots, A\_m$ は $(4r, \delta)$-restricted isometry propertyを満たすとする．ただし，$\delta = O(1/ (\kappa^3 \sqrt{r} \log^2 d))$
- 初期スケール $\alpha$ と学習率 $\eta$ を $d, r, \delta, \kappa$ に依存して適切に決める

このとき，$T$ が適切なオーダーの範囲に収まるならば，
$$
\lVert U\_T U\_T^\top - X^\* \rVert^2\_F \leq C \alpha \sqrt{d} / \kappa^2.
$$

### Theorem 1.2

- $ y = 1^\top q(U^{\*\top} x)$ という形の観測が得られるとする．ただし，$q$ は座標ごとの2次関数とする．これは3層ニューラルネットの特別な場合とみなせる．
- $x$ がGaussian designのとき，汎化誤差に関して Theorem 1.1と類似の結果が得られる．

## コメント

- RIPが重要らしい
- 2次ニューラルネットの場合のRIPは $x$ がGaussian designであるというところから来る
- 証明は，アルゴリズム全体を通して，「(a) $U\_t$ の $r + 1$ 次特異値が小さくとどまる」ということと「(b) $U\_t U\_t^\top$ の $1, \ldots, r$ 次特異値は真の $X^\*$ の特異値に適切なレートで近づく」ということを同時に言うらしい
